### PR TITLE
Refactor: extract common query/response test code

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

File: src/tests/conformance/query/ping_pong.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //! Contains test with ping queries.
-//!
+//! ZG-CONFORMANCE-003
```
Comment (ljedrz) [inline src/tests/conformance/query/ping_pong.rs:2]: I'm all for moving the spec test names somewhere above, but we should probably be consistent about it; @niklaslong?

File: src/tests/conformance/mod.rs [hunk 2]
```diff
@@ -1,2 +1,27 @@
+use crate::{
+    protocol::codecs::binary::{BinaryMessage, Payload},
+    setup::node::Node,
+    tools::synth_node::SyntheticNode,
+};
+
 mod handshake;
 mod query;
+
+async fn perform_query_response_test(payload: Payload, check: &dyn Fn(&BinaryMessage) -> bool) {
```
Comment (ljedrz) [inline src/tests/conformance/mod.rs:10]:
```suggestion
async fn perform_query_response_test(query_msg: Payload, response_check: &dyn Fn(&BinaryMessage) -> bool) {
```

File: src/tests/conformance/mod.rs [hunk 3]
```diff
@@ -1,2 +1,27 @@
...
+    // Start Ripple node
+    let mut node = Node::start_with_peers(vec![]).await.unwrap();
+
+    // Start synth node and connect to Ripple
+    let mut synth_node = SyntheticNode::start().await.unwrap();
+    synth_node.connect(node.addr()).await.unwrap();
+
+    // Send message if payload given
```
Comment (ljedrz) [inline src/tests/conformance/mod.rs:18]:
```suggestion
    // Send the query message
```

File: src/tests/conformance/mod.rs [hunk 4]
```diff
@@ -1,2 +1,27 @@
...
+    synth_node.unicast(node.addr(), payload).unwrap();
+
+    // Wait for required response
```
Comment (ljedrz) [inline src/tests/conformance/mod.rs:21]:
```suggestion
    // Wait for a response and perform the given check for it
```

File: src/tests/conformance/mod.rs (hunk 2 above)
Comment (lwojtow) [reply src/tests/conformance/mod.rs]: Done

File: src/tests/conformance/mod.rs (hunk 3 above)
Comment (lwojtow) [reply src/tests/conformance/mod.rs]: Done

File: src/tests/conformance/mod.rs (hunk 4 above)
Comment (lwojtow) [reply src/tests/conformance/mod.rs]: Done

File: src/tests/conformance/query/ping_pong.rs (hunk 1 above)
Comment (niklaslong) [reply src/tests/conformance/query/ping_pong.rs:2]: So long as the codebase is easily searchable and it is clear what cases the comment identifies in the file, I'm fine with either or even a combination.

**Review by ljedrz — APPROVED** [review]
LGTM :+1:

**Pull request merged**